src/lib.rs

```diff
@@ -1138,6 +1142,54 @@ pub trait Itertools : Iterator {
         adaptors::dedup_by_with_count(self, cmp)
     }
 
+    /// Return an iterator adaptor that produces elements that appear more than once during the
+    /// iteration. Duplicates are detected using hash and equality.
+    ///
+    /// The iterator is stable, returning the duplicate items in the order in which they occur in
+    /// the adapted iterator. Each duplicate item is returned exactly once. If an item appears more
+    /// than twice, the second item is the item retained and the rest are discarded.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let data = vec![10, 20, 30, 20, 40, 10, 50];
+    /// itertools::assert_equal(data.into_iter().duplicate(),
+    ///                         vec![20, 10]);
+    /// ```
+    #[cfg(feature = "use_std")]
+    fn duplicate(self) -> Duplicate<Self>
```

Comment:
I'm a bit iffy on the name. `duplicate`, in the context of a method name, reads as a verb to me, but this isn't a method that *duplicates* items. I'm hoping that pinning down the difference between `unique` and `duplicate` leads to a more fitting name.

Reply:
Hm, yeah it could be mistaken for a method that duplicates. Perhaps it could be named `duplicates`? Making it unambiguously a noun, hinting that this will produce the duplicate values?

Reply:
Actually, given [this comment](https://github.com/rust-itertools/itertools/pull/502#discussion_r559745443), `duplicate` sounds pretty fitting.

How about calling it `duplicates`, instead?

Reply:
Looks like we arrived at the same conclusion :) I'll rename them to `duplicates` and `duplicates_by`

Reply:
Don't forget to change the type names, too (`Duplicates` and `DuplicatesBy`)! (Gah. For some reason, *those* still read as verbs in my head...)

Reply:
For sure, I wouldn't leave some names as `duplicates` and some as `Duplicate`. What am I, some kind of a madman?? :joy: